### DISCUSSION TITLE
Access ContentDefinition.json file content

### DISCUSSION THREAD

**Discussion by sayyad84** [OrchardCMS/OrchardCore#15965] [Q&A]
Hi guys,
How can i access ContentDefinition.json file? i need access dropdown list text values that i defined them in text item.

**Comment by hishamco** [comment]
If you 're using `FileContentDefinitionStore` you will be able to get the `ContentDefinition.json` content

**Comment by deanmarcussen** [comment]
what you can do is access the content type definition using the `IContentDefintionManager`, find the `PartFieldDefinition` for the text field that is a Predefined List and call `var listSettings = Model.PartFieldDefinition.GetSettings<TextFieldPredefinedListEditorSettings>();`

see in the editor view also https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.ContentFields/Views/TextField-PredefinedList.Edit.cshtml

**Comment by Skrypt** [comment]
I think he meant from Liquid.

**Comment by deanmarcussen** [comment]
Hmm from liquid that will be harder.

But I exposed the `PartFieldDefintion` model recently for liquid, so we would just need a `get_settings` filter I would think

**Comment by Skrypt** [comment]
Is it documented?

**Comment by sayyad84** [comment]
> I think he meant from Liquid.

Thanks for reply.
yes, i need it to use in liquid templates.

**Comment by sebastienros** [comment]
I'd suggest you create a cshtml shape template instead for this part, and then inject it in a liquid page. You don't want to provide too much power in liquid files, that's the goal of liquid.
